server/api/lib/trees/regional_tree_permit.js

```diff
@@ -0,0 +1,233 @@
+				 File Could be found here: ${pathname}`);
+
+						const treePermits = await parseTreesXLS(pathname);
+						resolve(treePermits);
+					});
+				}
+				catch (err) {
+					Log.error(`Error fetching file ${url} :  ${err}`);
+					reject(err);
+				}
+				finally {
+					() => { clearTimeout(controllerTimeout); };
+				}
+			})();
+		});
+	}
+	catch (err) {
+		Log.error(err);
+		return Promise.reject();
+	}
+}
+
+async function saveNewTreePermits(treePermits) {
+	// Tree permits are published for objecctions for a period of 2 weeks. taking a 12 months
+	// buffer should be enough for human to remove those lines from the excel sheet.
+	//We're reading a the rows as a bulk and match them at compute time for performance.
+
+	if (treePermits.length == 0) return [];
+	// all tree permits in a chunk should be from the same regional office
+	const regionalOffice = treePermits[0].attributes[tpc.REGIONAL_OFFICE];
+	// this is the only timestamp format knex correcrtly work with 
+	const time_ago = moment().subtract(1, 'year').format('YYYY-MM-DDTHH:mm:ssZ');
+	const existingPermitsCompact = new Set();
+	await database.Knex(tpc.TREE_PERMIT_TABLE).where('updated_at', '>', time_ago.toString())
+		.andWhere(tpc.REGIONAL_OFFICE, regionalOffice)
+		.then(rows => {
+			for (let row of rows) {
+				const key_as_string =
+					[
+						row[tpc.PERMIT_NUMBER],
+						row[tpc.TREE_NAME],
+						row[tpc.NUMBER_OF_TREES],
+						row[tpc.END_DATE]
+					].join('_');
+				existingPermitsCompact.add(key_as_string);
+			}
+		})
+		.catch(function (error) { Log.error(error); });
+
+	const new_tree_permits = treePermits.map(tp => {
+		//if tp is not in the hash map of the existing one - add to the new ones
+		const compact_tp =
+			[
+				tp.attributes[tpc.PERMIT_NUMBER],
+				tp.attributes[tpc.TREE_NAME],
+				tp.attributes[tpc.NUMBER_OF_TREES],
+				tp.attributes[tpc.END_DATE]
+			].join('_');
+
+		if (tp.attributes[tpc.REGIONAL_OFFICE] == regionalOffice && !existingPermitsCompact.has(compact_tp)) {
+			Log.debug(`A new one! queued for saving ${compact_tp}`);
+			return tp; //original one, not compact
+		}
+	}).filter(Boolean); // remove undefined values
+	//save only the new ones
+	try { //TODO promise all or knex save bulk
+		new_tree_permits.map(async tp => {
+			Log.info(`Saving new tree permit: ${tp.attributes[tpc.PERMIT_NUMBER]} with ${tp.attributes[tpc.NUMBER_OF_TREES]} ${tp.attributes[tpc.TREE_NAME]} trees.`);
+			await tp.save();
+		});
+	}
+	catch (err) {
+		Log.error(err);
+	}
+	return new_tree_permits;
+}
+
+const parseTreesXLS = async (filename) => {
+	// hack
+	const sheetname = path.parse(filename).name.toLowerCase().includes('after') ? SHEET_AFTER : SHEET_BEFORE;
+	const workbook = xlsx.readFile(filename);
+	const sheet = workbook.Sheets[sheetname];
+	const sheet_json = xlsx.utils.sheet_to_json(sheet, { raw: false });
+	const treePermits = sheet_json.map(row => {
+		return new TreePermit(
+			{
+				[tpc.REGIONAL_OFFICE]: row['אזור'],
```

Comment:
Maybe we could save all hebrew titles at the const file, similar to the English consts. Easier when titles are changed in the excel format